kong/db/dao/init.lua

```diff
@@ -844,21 +937,33 @@ function DAO:each(size, options)
 
   if options ~= nil then
     validate_options_type(options)
+
+    if type(options.pagination) == "table" then
+      options.pagination = utils.table_merge(self.pagination, options.pagination)
+
+    else
+      options.pagination = self.pagination
+    end
+
+  else
+    options = {
+      pagination = self.pagination,
+    }
```

Comment:
a block like this seems to be added in four places — perhaps it would be better to make a local utility function to adjust the pagination option?

Reply:
change made.